### PR TITLE
ENG-9312: Move VoltProcedure deprecated functions elsewhere

### PR DESCRIPTION
**Pull request by jhugg**
This is a bit hackish as we still use these internally and for test a bit. I think my solution is a pretty safe one though. I have some additional work to take this a bit further, but it won't make it for 6.0. Still, 6.0 is the best time to remove these procs.

Has a pro pull as well.

### CODE REVIEW

File: tests/testprocs/org/voltdb_testprocs/regressionsuites/saverestore/GetTxnId.java [hunk 1]
```diff
@@ -35,7 +39,7 @@
         new SQLStmt("INSERT INTO PARTITION_TESTER VALUES (?, ?, ?, ?);");
 
     public VoltTable[] run(int partitioning) {
-        this.setAppStatusString(Long.toString(this.getVoltPrivateRealTransactionIdDontUseMe()));
+        this.setAppStatusString(Long.toString(DeprecatedProcedureAPIAccess.getVoltPrivateRealTransactionId(this)));
```
Comment (BillWhite) [inline tests/testprocs/org/voltdb_testprocs/regressionsuites/saverestore/GetTxnId.java]: Should this.setAppStatusString be just setAppStatusString?

**Comment by BillWhite** [comment]
Just one small comment.  You made these into static functions, right.  That seems safe.  You can't make them package or protected because they are used in inconvenient places.  So keeping them public seems ok.

File: tests/testprocs/org/voltdb_testprocs/regressionsuites/saverestore/GetTxnId.java (hunk 1 above)
Comment (jhugg) [inline tests/testprocs/org/voltdb_testprocs/regressionsuites/saverestore/GetTxnId.java]: Maybe? I saw a bunch of style changes I could make in code around where I made changes, but tried to change as little as possible for this pull.

Comment (BillWhite) [inline tests/testprocs/org/voltdb_testprocs/regressionsuites/saverestore/GetTxnId.java]: Well, this is the kind of think Paul comments on, so I mentioned it.  But I'm indifferent, as I reckon the compiler will know what to do with it.

**Pull request merged**